Sources/XMTPiOS/Conversations.swift

```diff
@@ -279,158 +441,84 @@
 																							   groupPinnedFrameUrl: pinnedFrameUrl,
                                                                                                customPermissionPolicySet: permissionPolicySet
-																   )).fromFFI(client: client)
+																   )).groupFromFFI(client: client)
 		try await client.contacts.allowGroups(groupIds: [group.id])
 		return group
 	}
-
-	/// Import a previously seen conversation.
-	/// See Conversation.toTopicData()
-	public func importTopicData(data: Xmtp_KeystoreApi_V1_TopicMap.TopicData) -> Conversation {
-		let conversation: Conversation
-		if !data.hasInvitation {
-			let sentAt = Date(timeIntervalSince1970: TimeInterval(data.createdNs / 1_000_000_000))
-			conversation = .v1(ConversationV1(client: client, peerAddress: data.peerAddress, sentAt: sentAt))
-		} else {
-			conversation = .v2(ConversationV2(
-				topic: data.invitation.topic,
-				keyMaterial: data.invitation.aes256GcmHkdfSha256.keyMaterial,
-				context: data.invitation.context,
-				peerAddress: data.peerAddress,
-				client: client,
-				createdAtNs: data.createdNs
-			))
-		}
-		Task {
-			await self.addConversation(conversation)
-		}
-		return conversation
-	}
-
-	public func listBatchMessages(topics: [String: Pagination?]) async throws -> [DecodedMessage] {
-		let requests = topics.map { topic, page in
-			makeQueryRequest(topic: topic, pagination: page)
-		}
-		/// The maximum number of requests permitted in a single batch call.
-		let maxQueryRequestsPerBatch = 50
-		let batches = requests.chunks(maxQueryRequestsPerBatch)
-			.map { requests in BatchQueryRequest.with { $0.requests = requests } }
-		var messages: [DecodedMessage] = []
-		// TODO: consider using a task group here for parallel batch calls
-		guard let apiClient = client.apiClient else {
-			throw ClientError.noV2Client("Error no V2 client initialized")
-		}
-		for batch in batches {
-			messages += try await apiClient.batchQuery(request: batch)
-				.responses.flatMap { res in
-					res.envelopes.compactMap { envelope in
-						let conversation = conversationsByTopic[envelope.contentTopic]
-						if conversation == nil {
-							print("discarding message, unknown conversation \(envelope)")
-							return nil
+	
+	public func streamAllConversationMessages() -> AsyncThrowingStream<DecodedMessage, Error> {
+		AsyncThrowingStream { continuation in
+			if (client.hasV2Client) {
+				continuation.finish(throwing: ConversationError.v2NotSupported("Only supported with V3 clients. Use streamAllMessages instead."))
+				return
+			}
+			let ffiStreamActor = FfiStreamActor()
+			let task = Task {
+				let stream = await self.client.v3Client?.conversations().streamAllMessages(
+					messageCallback: MessageCallback(client: self.client) { message in
+						guard !Task.isCancelled else {
+							continuation.finish()
+							Task {
+								await ffiStreamActor.endStream() // End the stream upon cancellation
+							}
+							return
```

Comment:
Not sure if we should try these suggestions since the streams seem to be working and it makes me a little nervous changing the way they previously worked since we've had a lot of issues in the past.